weed/storage/volume_checking.go

```diff
@@ -118,6 +120,9 @@ func verifyNeedleIntegrity(datFile backend.BackendStorageFile, v needle.Version,
 			return n.AppendAtNs, nil
 		}
 		if fileSize > fileTailOffset {
+			if (100 - (fileTailOffset * 100 / fileSize)) > dataLossLimitPerc {
```

Comment:
I could not following why the logic here can solve the problem.

Reply:
> I am not following why the logic here can solve the problem.

This is not exactly a solution to the problem, about the restriction on deleting data from dat files due to a broken idx
I want to avoid:
```
Truncate /data/bucket_254.dat from 421324863000 bytes to 234354667858 bytes!
```
If you don’t do Truncate, the volume will go into read-only mode and you can rebuild the index manually using a `weed fix`